compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -39,35 +39,33 @@ pub struct OpaqueTypeDecl<'tcx> {
 }
 
 impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
-    /// This is a backwards compatibility hack to prevent breaking changes from
-    /// lazy TAIT around RPIT handling.
     pub fn replace_opaque_types_with_inference_vars<T: TypeFoldable<'tcx>>(
         &self,
         value: T,
         body_id: HirId,
         span: Span,
         param_env: ty::ParamEnv<'tcx>,
+        // This is a backwards compatibility hack to prevent breaking changes from
+        // lazy TAIT around RPIT handling.
+        only_for_rpit: bool,
```

Comment:
can you use a custom enum for that :thinking: seeing `true` or `false` is somewhat confusing

Reply:
it was in fact so confusing that porting it over to the enum without looking properly caused a bug :D